tools/esdlcomp/esdlcomp.h

```diff
@@ -892,12 +892,16 @@ class EspMessageInfo
     {
         if (!xsdgrouptype)
         {
-            xsdgrouptype = strdup(getMetaString("xsd_group_type", "\"all\""));
-            if (*xsdgrouptype == '\"')
-                xsdgrouptype++;
```

Comment:
This code looked very dodgy to me - assuming that someone somewhere eventually frees this string it's going to end up passing something to free that was not allocated on the heap.